public/app/features/panel/panellinks/link_srv.ts

```diff
@@ -1,15 +1,42 @@
-import angular from 'angular';
 import _ from 'lodash';
+import { TimeSrv } from 'app/features/dashboard/services/TimeSrv';
+import { TemplateSrv } from 'app/features/templating/template_srv';
+import coreModule from 'app/core/core_module';
 import { appendQueryToUrl, toUrlParams } from 'app/core/utils/url';
-import { PanelDrillDownLink, KeyValue } from '@grafana/ui';
+import { PanelDrillDownLink, KeyValue, ScopedVars, DateTime, dateTime } from '@grafana/ui';
+import { TimeSeriesValue, transformAbsoluteTimeRange } from '@grafana/ui';
 
 const DrilldownLinkBuiltInVars = {
   keepTime: '__urlTimeRange',
   includeVars: '__allVariables',
+  seriesName: '__seriesName',
+  dataPointTs: '__dataPointTs',
 };
-export class LinkSrv {
+
+// Represents factor by which the original time range will be narrowed down when
+// using drilldown link
+const DrilldownLinkDataPointRangeFactor = 0.5;
+
+type LinkTarget = '_blank' | '_self';
+
+interface LinkModel {
+  url: string;
+  title: string;
+  target: LinkTarget;
+}
+
+interface LinkDataPoint {
+  datapoint: TimeSeriesValue[];
+  seriesLabel: string;
```

Comment:
seriesName I think is better to not confuse this with SeriesData labels & label values

Reply:
how about fieldName?  these values are filled up from TimeSeries | TableData